restore.js

```diff
@@ -0,0 +1,74 @@
+import readline from 'readline';
+
+import * as firebase from 'firebase';
+import 'firebase/firestore';
+import fs from 'fs';
+
+function restoreDatabase() {
+    firebase.initializeApp(require('./firebase.test.config.json'));
+    const auth = firebase.auth();
+    const db = firebase.firestore();
+
+    const queries = {
+        'operations': db.collection('operations'),
+        'members': db.collection('members'),
+        'relations': db.collection('relations'),
+        'user_data': db.collection('user_data')
+    };
+
+    let queriesCompleted = 0;
+    const queriesToComplete = Object.keys(queries).length;
+
+    Object.entries(queries).forEach((nmQry) => {
+        let [name, query] = nmQry;
+        const backupFilename = `./backup.${name}.json`;
+
+        new Promise((resolve) => {
+            fs.readFile(backupFilename, 'utf8', function(err, data) {
+                if (err) throw err;
+                const objects = [];
+                resolve(data.split('\n').map((object) => JSON.parse(object)));
+            });
+        }).then((fileJson) => {
+            const totalToCreate = fileJson.length;
+            let created = 0;
+            fileJson.forEach((object) => {
+                const { id, data } = object;
+                query.doc(id).set(data).then(() => {
+                    if (++created == totalToCreate) {
+                        console.log(`Restored ${totalToCreate} objects to the ${name} collection.`);
```

Comment:
TMLI - Too Many Levels of Indentation - https://web.stanford.edu/~ouster/cgi-bin/cs190-winter18/reviewCodes.php